### DISCUSSION TITLE
I randomly get OpenSSL SSL_connect error

### DISCUSSION THREAD

**Discussion by aalinia** [kreait/firebase-php#640] [Q&A] [answered]
I randomly get `OpenSSL SSL_connect` errors (avg once in 10-15 request) that are the same with different target URLs (`oauth2.googleapis.com:443` and `fcm.googleapis.com:443`).
Any idea about why it happens or how to fix it?

Errors:
`PHP message: [critical] Uncaught PHP Exception Kreait\Firebase\Exception\Messaging\MessagingError: "cURL error 35: OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to oauth2.googleapis.com:443  (see https://curl.haxx.se/libcurl/c/libcurl-errors.html)" at /srv/api/vendor/kreait/firebase-php/src/Firebase/Exception/MessagingApiExceptionConverter.php line 94`
or
`PHP message: [critical] Uncaught PHP Exception Kreait\Firebase\Exception\Messaging\MessagingError: "cURL error 35: OpenSSL SSL_connect: SSL_ERROR_SYSCALL in connection to fcm.googleapis.com:443  (see https://curl.haxx.se/libcurl/c/libcurl-errors.html)" at /srv/api/vendor/kreait/firebase-php/src/Firebase/Exception/MessagingApiExc
eptionConverter.php line 94`

**Comment by jeromegamez** [comment, marked as answer]
curl, OpenSSL and connectivity errors in general are the worst... unfortunately I can't give more recommendations than the usual: try updating the cURL and OpenSSL versions on the system in question, ensure your PHP version is supported and up to date and uses current root certificates (https://firebase-php.readthedocs.io/en/5.x/troubleshooting.html#curl-error-xx), check the Internet Connection and proxies (if you use any).

It's especially difficult since you said that it works most of the time and "just" randomly fails - this could mean that all I mentioned above is probably fine (but you should check anyway). If at home, sometimes a computer and/or router restart helps - if this happens on a remote environment, another approach could be to check if the problems occur as well if you use another environment in another data center or with a different service provider.

Last but not least: it could be that the problem lies within the Firebase Services themselves, in that case the Probleme might resolve themselves after some time (hopefully quickly) 🤞

**Reply by aalinia** [reply]
@jeromegamez, Thanks for your quick answer

I have checked everything and they look fine to me. IMO it can be because of either the server(service provider) network configuration (which is hard to test with another provider now because it is our production) or firebase services.
